ui/app/AppLayouts/Wallet/popups/swap/SwapModalAdaptor.qml

```diff
@@ -32,8 +32,11 @@ QObject {
                 expectedRoles: ["address"]
             },
             FastExpressionRole {
+                function getFromToken() {
+                    return ModelUtils.getByKey(root.walletAssetsStore.walletTokensStore.plainTokensBySymbolModel, "key", root.swapFormData.fromTokensKey)
```

Comment:
This value is not depending on `model` data and it is the same static value. Maybe it makes sense to be cached on `root.swapFormData.fromTokensKey` change

Reply:
Yup, the extra roles are not cheap